.github/cargo_local_patch.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+import toml
+import os
+
+def modify_dependency(details, new_path):
+    if 'path' not in details:
+        details['path'] = new_path
+        details.pop('version', None)
+        details.pop('git', None)
+        details.pop('rev', None)
+        details.pop('branch', None)
+
+def process_dependencies(dependencies, base_path, dep_path_mapping):
+    for dep, details in dependencies.items():
+        # Check if the current dependency is in our mapping
+        if dep in dep_path_mapping:
+            # Construct the new path using the base_path and the relative path from the mapping
+            new_path = os.path.join(base_path, dep_path_mapping[dep])
+            modify_dependency(details, new_path)
+
+
+def calculate_base_path(cargo_toml_path):
+    depth = len(os.path.normpath(cargo_toml_path).split(os.sep)) - 1
+    if depth == 0:
+        return "."
+    else:
+        return os.path.join(*[".."] * depth)
+
+def main(cargo_toml_path, dep_path_mapping):
+    base_path = calculate_base_path(cargo_toml_path)
+
+    with open(cargo_toml_path, 'r') as file:
+        data = toml.load(file)
+
+    if 'dependencies' in data:
+        dependencies = data['dependencies']
```

Comment:
Some of the entries in 'dependencies' will be a string instead of a dict (e.g. `anyhow = "1.0"`), which will end up causing this to fail. Maybe that's ok though 🤷

Reply:
I'll add a functionality to address that case